glommio/src/io/directory.rs

```diff
@@ -87,6 +88,25 @@ impl Directory {
         Self::sync_open(&path)
     }
 
+    /// Creates a file under this directory, returns a DMA file
+    ///
+    /// NOTE: Path must not contain directories and just be a file name
+    pub async fn create_file<P: AsRef<Path>>(&self, path: P) -> io::Result<DmaFile> {
```

Comment:
This should probably accept a `AsRef<OsStr>` instead of `AsRef<Path>`, as it only expects a single path component.

Reply:
But doesn't this make it harder to validate if this is indeed a relative path ?